leetcodePython/SlidingWindow/slidingwindow_2781.py

```diff
@@ -0,0 +1,18 @@
+from typing import List
+
+
+class Solution:
+    def longestValidSubstring(self, word: str, forbidden: List[str]) -> int:
+        forbidden_set = set(forbidden)
+
+        # forbidden_max_length = max([len(f) for f in forbidden_set])
+        forbidden_max_length = 10
+        max_length = 0
+        valid_left = 0
+        for right in range(len(word)):
+            left_start = max(valid_left, right - forbidden_max_length)
```

Comment:
1. We only start from when the left is valid
2. The forbidden have limit size (10). We only need to start from this range before right.